src/NServiceBus.Transport.RabbitMQ/Connection/ConnectionFactory.cs

```diff
@@ -82,12 +83,14 @@ public ConnectionFactory(string endpointName, ConnectionConfiguration connection
 
         public IConnection CreateAdministrationConnection() => CreateConnection($"{endpointName} Administration", false);
 
-        public IConnection CreateConnection(string connectionName, bool automaticRecoveryEnabled = true)
+        public IConnection CreateConnection(string connectionName, bool automaticRecoveryEnabled = true, int consumerDispatchConcurrency = 1)
         {
             lock (lockObject)
             {
                 connectionFactory.AutomaticRecoveryEnabled = automaticRecoveryEnabled;
                 connectionFactory.ClientProperties["connected"] = DateTime.Now.ToString("G");
+                // it is OK to modify as long as we are under a lock
+                connectionFactory.ConsumerDispatchConcurrency = consumerDispatchConcurrency;
```

Comment:
I felt given there are assumptions that it is OK to modify the connection factory settings per connection creation and we already use a lock here I decided to just piggy back on this for now

Reply:
Yeah this seems fine to do here. Not sure it needs the comment since we already have other things being modified inside the lock.